soh/src/overlays/actors/ovl_En_Xc/z_en_xc.c

```diff
@@ -283,17 +283,32 @@ void func_80B3CA38(EnXc* this, GlobalContext* globalCtx) {
     }
 }
 
+void GivePlayerRandoRewardSheikSong(EnXc* sheik, GlobalContext* globalCtx, RandomizerCheck check, int sheikType) {
+    if (sheik->actor.parent != NULL && sheik->actor.parent->id == GET_PLAYER(globalCtx)->actor.id &&
+        !(gSaveContext.eventChkInf[5] & sheikType)) {
+        gSaveContext.eventChkInf[5] |= sheikType;
+    } else if (!(gSaveContext.eventChkInf[5] & sheikType)) {
+        GetItemID getItemId = GetRandomizedItemIdFromKnownCheck(check, GI_MINUET_OF_FOREST);
```

Comment:
another note that won't block merging, this call happens for more than just minuet, i don't expect it to be a problem but i'm thinking i'll refactor this so GivePlayerRandoRewardSheikSong also takes in a GetItemID